.github/workflows/ci.yml

```
@@ -0,0 +1,181 @@
+    tags:
+      - v*
+    paths-ignore:
+      - 'README.md'
+jobs:
+  fmt:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          default: true
+          components: rustfmt
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Check Formatting
+        uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --all --check
+
+  clippy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          default: true
+          components: clippy
+
+      - name: Cache
+        uses: Swatinem/rust-cache@v2
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Clippy with features
+        uses: actions-rs/clippy-check@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          args: --all-features --all-targets -- -D warnings
+
+      - name: Clippy without features
+        uses: actions-rs/clippy-check@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          args: --all-targets -- -D warnings
+
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        # We want newer versions than 'latest' here to have current wasm-opt
+        os: ["ubuntu-22.04", "macos-12", "windows-2022"]
+    runs-on: ${{ matrix.os }}
+    env:
+      RUST_BACKTRACE: full
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install wasm-opt
+        if: ${{ runner.os == 'Linux' }}
+        run: sudo apt-get install binaryen
+
+      - name: Install wasm-opt
+        if: ${{ runner.os == 'macOS' }}
+        run: brew install binaryen
+
+      - name: Install wasm-opt
+        if: ${{ runner.os == 'Windows' }}
+        uses: engineerd/configurator@v0.0.8
+        with:
+          name: wasm-opt.exe
+          url: https://github.com/WebAssembly/binaryen/releases/download/version_109/binaryen-version_109-x86_64-windows.tar.gz
+          pathInArchive: binaryen-version_109/bin/wasm-opt.exe
+
+      - name: Install stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          default: true
+          target: wasm32-unknown-unknown
+          components: rust-src
+
+      - name: Cache
+        uses: Swatinem/rust-cache@v2
+
+      - name: Install cargo-dylint
+        uses: baptiste0928/cargo-install@v1
+        with:
+          crate: cargo-dylint
+          version: 2
+
+      - name: Install dylint-link
+        uses: baptiste0928/cargo-install@v1
+        with:
+          crate: dylint-link
+          version: 2
+
+      - name: Run Tests
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --workspace --all-features -- --test-threads=1
```

```suggestion
          args: --workspace --all-features
```
Don't know how many cores the GHA machines have but this will surely speed things up. I have successfully run tests in parallel locally.

Possibly the original reason for this is no longer an issue,

---

Does not really speed up because the test itself compile stuff which is multi threaded. All the other tests run in seconds. 

> Possibly the original reason for this is no longer an issue,

Yes. Original reason was the linting which is no longer run for those tests. However the output will be completely mangled with multi hreaded tests. I would rather leave it single for this reason.